frontend/packages/console-app/src/components/guided-tours/QuickStartSection.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { last } from 'lodash';
+import { Link } from 'react-router-dom';
+import { Button } from '@patternfly/react-core';
+import { GuidedTourItem, GuidedTourState, GuidedTourTaskItem } from './utils/guided-tour-typings';
+import { GuidedTourStatus, TaskState } from './utils/guided-tour-status';
+import QuickStartTask from './QuickStartTask';
+import TaskHeader from './TaskHeader';
+import './QuickStartSection.scss';
+
+type QuickStartSectionProps = {
+  quickStart: GuidedTourItem;
+  quickStartState: GuidedTourState;
+  nextCallback?: () => void;
+  reviewCallback?: (reviewState: boolean) => void;
+  prevCallback?: () => void;
+};
+
+enum PrimaryButtonText {
+  START = 'Start Tour',
+  NEXT = 'Next',
+  CLOSE = 'Close',
+}
+
+const QuickStartSection: React.FC<QuickStartSectionProps> = ({
+  quickStart,
+  quickStartState,
+  nextCallback,
+  reviewCallback,
+  prevCallback,
+}) => {
+  const { introduction, tasks, conclusion } = quickStart;
+  const { tourState, tasks: tasksWithStates } = quickStartState;
+
+  const subtitle = (task) =>
+    task.title &&
+    `${tasks.findIndex((taskElement) => taskElement.title === task.title) + 1} of ${tasks.length}`;
+
+  const isFinalScreen: boolean =
+    tourState === GuidedTourStatus.IN_PROGRESS &&
+    tasksWithStates?.length === tasks.length &&
+    (last(tasksWithStates).taskState === TaskState.SUCCESS ||
+      last(tasksWithStates).taskState === TaskState.FAILED);
+
+  const getPrimaryButtonText = (): PrimaryButtonText => {
+    if (isFinalScreen) {
+      return PrimaryButtonText.CLOSE;
+    }
+    if (tourState === GuidedTourStatus.IN_PROGRESS) {
+      return PrimaryButtonText.NEXT;
+    }
+    return PrimaryButtonText.START;
+  };
+
+  const renderTasks = (): GuidedTourTaskItem[] => {
+    if (tourState === GuidedTourStatus.NOT_STARTED) {
+      return [{ description: introduction }];
+    }
+    if (isFinalScreen) {
+      return [...tasksWithStates, ...[{ description: conclusion }]];
+    }
+    return tasksWithStates;
+  };
```

Comment:
Same goes for this logic. You're recalculating all the steps that need to be rendered based on the state of the tour. instead it should just render the current step out of the steps array. Based on the state `NOT_STARTED` or `COMPLETED` you should render `Introduction` or `Conclusion` step of the tour.